### PR TITLE
Dynamodb table replica support

### LINKED ISSUE #41: Global Table regions
Does this module support creating replica table(s) in another region? How?

### PR DESCRIPTION
**Pull request by mwarkentin**
## what
* This feature will allow setting replica configurations for aws_dynamodb_table resource.

## why
* According to the note at [AWS provider v3.4.0 docs](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/dynamodb_global_table), it's recommended to use replica configuration instead of aws_dynamodb_global_table resource and there is a lack of that functionality in this module.

## references
* Closes #41
* Replaces #66, #82

### CODE REVIEW

**Comment by mwarkentin** [comment]
FYI, I haven't tested these changes myself yet, working on getting a terraform 0.13 environment set up to test (we're in the middle of upgrading from 12 -> 13 right now).

File: main.tf [hunk 1]
```diff
@@ -51,8 +51,9 @@ resource "aws_dynamodb_table" "default" {
   write_capacity   = var.autoscale_min_write_capacity
   hash_key         = var.hash_key
   range_key        = var.range_key
-  stream_enabled   = var.enable_streams
-  stream_view_type = var.enable_streams ? var.stream_view_type : ""
+  stream_enabled   = length(var.replicas) > 0 ? true : var.stream_enabled
```
Comment (SweetOps) [inline main.tf:54]:
```suggestion
  stream_enabled   = length(var.replicas) > 0 ? true : var.enable_streams
```

to fix 
```
TestExamplesComplete 2021-04-29T13:45:38Z command.go:121: Error: Reference to undeclared input variable
94
TestExamplesComplete 2021-04-29T13:45:38Z command.go:121: 
95
TestExamplesComplete 2021-04-29T13:45:38Z command.go:121:   on ../../main.tf line 54, in resource "aws_dynamodb_table" "default":
96
TestExamplesComplete 2021-04-29T13:45:38Z command.go:121:   54:   stream_enabled   = length(var.replicas) > 0 ? true : var.stream_enabled
97
TestExamplesComplete 2021-04-29T13:45:38Z command.go:121: 
98
TestExamplesComplete 2021-04-29T13:45:38Z command.go:121: An input variable with the name "stream_enabled" has not been declared.
```

File: main.tf [hunk 2]
```diff
@@ -51,8 +51,9 @@ resource "aws_dynamodb_table" "default" {
   write_capacity   = var.autoscale_min_write_capacity
   hash_key         = var.hash_key
   range_key        = var.range_key
-  stream_enabled   = var.enable_streams
-  stream_view_type = var.enable_streams ? var.stream_view_type : ""
+  stream_enabled   = length(var.replicas) > 0 ? true : var.enable_streams
+  stream_view_type = length(var.replicas) > 0 || var.enable_streams ? var.stream_view_type : ""
```
Comment (mwarkentin) [inline main.tf:55]: I believe for global tables `stream_view_type` needs to be `NEW_AND_OLD_IMAGES` - is that something we want to configure automatically or just document and expect the user to pass in?

**Comment by mwarkentin** [comment]
Looks like there's some `billing_mode` requirements as well:

```
Error: error initially creating DynamoDB Table (wave-devmw-multi-region) replicas: error creating DynamoDB Table (wave-devmw-multi-region) replica (us-east-2): ValidationException: Table write capacity should either be Pay-Per-Request or AutoScaled.
```

**Comment by mwarkentin** [comment]
@SweetOps ready for another look whenever you're around. I've tested:

* DynamoDB tables creation works in both single-region and multi-region mode ([with a caveat](https://github.com/cloudposse/terraform-aws-dynamodb/pull/83#issuecomment-829343491))
* Single-region table can be converted to multi-region, and back again

**Review by SweetOps — APPROVED** [review]

**Comment by mwarkentin** [comment]
@SweetOps thanks!

**Pull request merged**